### PR TITLE
Refactor

### LINKED ISSUE #40: Refactor the build process
CogStack-Pipeline/build.gradle 
seems to doing a lot of extraneous things like building containers. 

- [x] cleanup container building code
- [x] Use docker-compose to build the containers
- [x] odd scripts
- [x] config file generation
- [x] Move old scripts to deprecated and will be deleted in the future builds
- [x] Add initial travis build
- [x] Configure Travis for CI. Will require admin access to github repo

### PR DESCRIPTION
**Pull request by yatharthranjan**
@afolarin i have also added a simple travis build but could not configure the travis CI as I don't have admin access to the repo. Can you please review this and also configure the travis CI ? Thanks

closes #40 

- [x] cleanup container building code
- [x] Use docker-compose instead of gradle to build the containers
- [x] odd scripts
- [x] config file generation
- [x] Move old scripts to deprecated and will be deleted in the future builds
- [x] Add initial travis build
- [x] Configure Travis for CI. Will require admin access to github repo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,15 +56,7 @@ sudo sysctl -w vm.max_map_count=262144
...
-From the CogStack top level directory:
-
-```
-gradlew buildSimpleContainers
-```
-
-Assuming the containers have been built successfully, simply navigate to
+Now simply navigate to
 ```
 cd docker-cogstack/compose-ymls/simple/
```
Comment (afolarin) [inline README.md:61]: don't think all these containers are cloud built yet

File: docker-cogstack/compose-ymls/cogstack-clust/docker-compose.yml [hunk 2]
```diff
@@ -6,6 +6,7 @@ services:
 # Postgres container for spring batch / cogstack testing                    #
 #---------------------------------------------------------------------------#
   postgres:
+    build: ../../postgres
```
Comment (afolarin) [inline docker-cogstack/compose-ymls/cogstack-clust/docker-compose.yml]: postgres and cogstack-pipeline are on automated builds

File: gradle/buildcontainer.gradle [hunk 3]
```diff
@@ -0,0 +1,44 @@
+// Deprecated. To be removed
```
Comment (afolarin) [inline gradle/buildcontainer.gradle:1]: agreed, we definitely don't want gradle doing the container builds.

**Review by afolarin — COMMENTED** [review]
also let me cc/ @Honghan

File: README.md (hunk 1 above)
Comment (yatharthranjan) [reply README.md:61]: Yes but i have specified build context in the docker-compose.yml file so should automatically build before running

File: docker-cogstack/compose-ymls/cogstack-clust/docker-compose.yml (hunk 2 above)
Comment (yatharthranjan) [reply docker-cogstack/compose-ymls/cogstack-clust/docker-compose.yml]: Okay cool. Did not know that. Will remove it.

File: gradle/buildcontainer.gradle (hunk 3 above)
Comment (yatharthranjan) [reply gradle/buildcontainer.gradle:1]: Yes these tasks will not be executed by default in the build process. Only if explicitly use the **run** profile while running a gradle build using -Dorg.gradle.project.profile=run

File: docker-cogstack/compose-ymls/cogstack-clust/docker-compose.yml (hunk 2 above)
Comment (afolarin) [reply docker-cogstack/compose-ymls/cogstack-clust/docker-compose.yml]: Think we should have different compose files for the difference container deployment options

**Review by afolarin — APPROVED** [review]
LGTM -- I think we can merge, I'll add a few comments to the following PR #47

**Pull request merged**